test/e2e/gardener/context.go

```diff
@@ -244,3 +245,48 @@ func (t *TestContext) ForSeed(seed *gardencorev1beta1.Seed) *SeedContext {
 
 	return s
 }
+
+// ManagedSeedContext is a test case-specific TestContext that carries test state and helpers through multiple steps of the
+// same test case, i.e., within the same ordered container.
+// Accordingly, ManagedSeedContext values must not be reused across multiple test cases (ordered containers). Make sure to
+// declare ManagedSeedContext variables within the ordered container and initialize them during ginkgo tree construction,
+// e.g., in a BeforeTestSetup node or when invoking a shared `test` func.
+//
+// A ManagedSeedContext can be initialized using TestContext.ForManagedSeed.
+type ManagedSeedContext struct {
+	TestContext
+
+	// ManagedSeed object the test is working with
+	ManagedSeed *seedmanagementv1alpha1.ManagedSeed
+
+	// ShootContext object the managed seed is referencing
+	ShootContext ShootContext
+
+	// Seed object the managed seed is referencing
+	SeedContext SeedContext
+}
+
+// ForManagedSeed copies the receiver ShootContext for deriving a ManagedSeedContext.
+func (t *TestContext) ForManagedSeed(managedseed *seedmanagementv1alpha1.ManagedSeed) *ManagedSeedContext {
+	ms := &ManagedSeedContext{
+		TestContext: *t,
+		ManagedSeed: managedseed,
+	}
+	t.Log = t.Log.WithValues("managedseed", client.ObjectKeyFromObject(managedseed))
+
+	return ms
+}
+
+// WithShoot initialized a ShootContext and adds it to the ManagedSeedContext
+// This function is intended to be used for the shoot the managed seed is "based" on
+func (s *ManagedSeedContext) WithShoot(shoot *gardencorev1beta1.Shoot) *ManagedSeedContext {
+	s.ShootContext = *s.ForShoot(shoot)
```

Comment:
Should we make `{Shoot,Seed}Context` pointers?

Reply:
Having a pointer suggests that it might be `nil`, which it is until `WithShoot` or `WithSeed` is called. Also, calls like `ItShouldCreateShoot(s.ShootContext)` feel more intuitive when `ShootContext` already is a pointer.

Reply:
I agree with you, will change it :)

Reply:
In order to prevent nil pointers being passed to functions, I changed the way the test contexts get initialized.

Reply:
Thanks, I like it :+1: